### ISSUE TITLE
Version release docs

### ISSUE THREAD

**Issue by devinrsmith** [deephaven/deephaven-core#4785]
Right now `docs-ci.yml` runs on every merge to main. We should update the release process to publish the documents into a separately versioned folder on the target machine. We should also consider uploading these artifacts as official release artifacts (as opposed to simply rsyncing them to the host).

**Comment by stanbrub** [comment]
When you say "official release artifacts" do you mean as Assets like here: https://github.com/deephaven/deephaven-core/releases/tag/v0.30.1

**Comment by stanbrub** [comment]
Should we split the _official release artifacts_ part into a separate ticket?  If I understand what you mean, that one seems like a quicker task than the Deploy.  I already have something for DocsCI that looks like the following (though it would still be a manual upload like the other assets):
![Screenshot from 2023-11-16 13-05-46](https://github.com/deephaven/deephaven-core/assets/120737309/8fa84420-d63c-4e6a-91d5-9553c1e35bf8)

**Comment by devinrsmith** [comment]
Yes - the idea would be to upload these as official release "assets" as well. Feel free to create separate ticket.
